part1/stages/Check-GPU

```diff
@@ -56,4 +56,66 @@ if [ ${GPU_OK} -ne 0 ] ; then
     echo "Ignoring unrecognised graphics device.">&2
 fi
 
+# Sourced from https://stackoverflow.com/a/8811800
+# BusyBox capable substring search
+# Try to remove the search substring from the search target string
+# If the stripped string is the same as the original target, substing was not found
+contains() {
+    local string="$1"
+    local substring="$2"
+    [ "${string#*$substring}" != "$string" ]
+}
+
+check_vga()
+{
+    OLD_IFS=$IFS
+    IFS="
+"
+    local onboard_pci="$(lspci -s 00: -d ::0300 | awk '{print $1}')"
+    for boot_vga in $(find /sys/ | grep boot_vga); do
```

Comment:
`boot_vga` is a file, so why not use `find /sys -name boot_vga`? That would exclude (unlikely) false matches that happen to contain that string. IMO it's also safe to expect the result to be free of spaces, removing the need for IFS manipulation, but better safe than sorry I guess.

Reply:
+1

@jean-edouard solution will ensure there is never a false positive match versus relying on the unlikely possibility.